paramiko/pkey.py

```diff
@@ -168,17 +168,74 @@ def can_sign(self):
         """
         return False
 
-    def get_fingerprint(self):
+    def get_sha256_fingerprint(self):
+        """
+        Returns an openssh SHA256 fingerprint of the public part of this key.
+        SHA256 fingerprint is calculated by base64 encoding the sha256 digest
+        of the key and then removing the base64 trailing == chars
+        Nothing secret is revealed.
+
+        :return:
+            a 43-byte `string <str>` (base64) of the SHA256 fingerprint,
+            in SSH format.
+        """
+        return base64.b64encode(b(sha256(self.asbytes()).digest()))[:-1]
+
+    def get_sha512_fingerprint(self):
+        """
+        Returns an openssh SHA512 fingerprint of the public part of this key.
+        SHA512 fingerprint is calculated by base64 encoding the sha512 digest
+        of the key and then removing the two base64 trailing == chars
+        Nothing secret is revealed.
+
+        :return:
+            a 86-byte `string <str>` (base64) of the SHA512 fingerprint,
+            in SSH format.
+        """
+        return base64.b64encode(b(sha512(self.asbytes()).digest()))[:-2]
+
+    def get_md5_fingerprint(self):
         """
         Return an MD5 fingerprint of the public part of this key.  Nothing
-        secret is revealed.
+        Nothing secret is revealed.
 
         :return:
             a 16-byte `string <str>` (binary) of the MD5 fingerprint, in SSH
             format.
         """
         return md5(self.asbytes()).digest()
 
+    def get_available_fingerprint_methods(self):
+        """
+        Return a dictionary of available hash algorithms and  their method
+        callbacks
```

Comment:
```suggestion
        callbacks.
```

Reply:
Please let me know when you finish these review comments. I hate to create multiple commits for such minor remarks as missing `.` in docstrings